dialects/direct/passes/cps2ds.cpp

```diff
@@ -88,15 +113,36 @@ const Def* CPS2DS::rewrite_body_(const Def* def) {
                             inst_ret_ty = ret_ty;
                         }
 
+                        auto new_name = *curr_lam_->sym();
+                        // append _cps_cont
+                        // if name contains _cps_cont append _1
+                        // if it contains _[n] append _[n+1]
+                        std::string append = "_cps_cont";
+                        auto pos           = new_name.find(append);
+                        if (pos != std::string::npos) {
+                            auto num = new_name.substr(pos + append.size());
+                            if (num.empty()) {
+                                new_name += "_1";
+                            } else {
+                                num      = num.substr(1);
+                                num      = std::to_string(std::stoi(num) + 1);
+                                new_name = new_name.substr(0, pos + append.size()) + "_" + num;
+                            }
+                        } else {
+                            new_name += append;
+                        }
```

Comment:
I think this is a useful idiom that we want to factor out into a utility function. But for now, we can keep it like this.